src/cairoWriter.ts

```diff
@@ -685,7 +685,8 @@ class IdentifierWriter extends CairoASTNodeWriter {
       isDynamicCallDataArray(getNodeType(node, this.ast.compilerVersion)) &&
       ((node.getClosestParentByType(Return) !== undefined &&
         node.getClosestParentByType(FunctionDefinition)?.visibility ===
-          FunctionVisibility.External) ||
+          FunctionVisibility.External &&
+        node.getClosestParentByType(IndexAccess) === undefined) ||
```

Comment:
This will likely need changing in the calldata array length PR, but since that's the only case I think, it's fine for now